src/main/src/main/java/org/geoserver/config/GeoServerReinitializer.java

```diff
@@ -0,0 +1,18 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.config;
+
+
+/**
+ * Extension point interface for initializing based on configuration.
+ * Intended to be run both upon configuration initialization, and upon reload
+ *
+ */
+public interface GeoServerReinitializer extends GeoServerInitializer { 
+
+    default void reinitialize( GeoServer geoServer ) throws Exception {
+        initialize(geoServer);
+    };
```

Comment:
Thanks, that is what I had in mind, can we get a javadoc (since this is a public method).